flask/helpers.py

```diff
@@ -575,12 +585,21 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
             warn('Access %s failed, maybe it does not exist, so ignore etags in '
                  'headers' % filename, stacklevel=2)
 
-        if conditional:
+    if conditional:
+        try:
+            # Added in werkzeug 0.12
+            rv = rv.make_conditional(request, accept_ranges=True,
+                                     complete_length=fsize)
+        except TypeError:
             rv = rv.make_conditional(request)
-            # make sure we don't send x-sendfile for servers that
-            # ignore the 304 status code for x-sendfile.
-            if rv.status_code == 304:
-                rv.headers.pop('x-sendfile', None)
+        except RequestedRangeNotSatisfiable:
+            if close_on_416:
```

Comment:
My point is that we should do this regardless of whether we opened the file ourselves or got it passed as parameter.